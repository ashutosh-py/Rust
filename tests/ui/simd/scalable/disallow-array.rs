#![feature(repr_simd, repr_scalable)]

#[repr(simd, scalable(4))]
pub struct ScalableSimdFloat {
    _ty: [f32]
}

fn main() {
    let x: [ScalableSimdFloat; 2]; //~ ERROR E0277
}
