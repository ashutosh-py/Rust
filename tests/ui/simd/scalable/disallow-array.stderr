error[E0277]: the size for values of type `[f32]` cannot be known at compilation time
  --> $DIR/disallow-array.rs:9:12
   |
LL |     let x: [ScalableSimdFloat; 2];
   |            ^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: within `ScalableSimdFloat`, the trait `Sized` is not implemented for `[f32]`, which is required by `ScalableSimdFloat: Sized`
note: required because it appears within the type `ScalableSimdFloat`
  --> $DIR/disallow-array.rs:4:12
   |
LL | pub struct ScalableSimdFloat {
   |            ^^^^^^^^^^^^^^^^^
   = note: slice and array elements must have `Sized` type

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
